stips/utilities/makePSF.py

```diff
@@ -0,0 +1,318 @@
+    if (dd > 4.0):
+        rpsf_phot = ( (1-fx)*(1-fy)*psf[iy  ,ix  ]
+                    + ( fx )*(1-fy)*psf[iy  ,ix+1]
+                    + (1-fx)*( fy )*psf[iy+1,ix  ]
+                    + ( fx )*( fy )*psf[iy+1,ix+1])
+        return rpsf_phot
+
+    # Bi-cubic interpolation
+    A1 =  psf[iy  ,ix  ]
+    B1 = (psf[iy  ,ix+1]-psf[iy  ,ix-1])/2
+    C1 = (psf[iy+1,ix  ]-psf[iy-1,ix  ])/2
+    D1 = (psf[iy  ,ix+1]+psf[iy  ,ix-1]-2*A1)/2
+    F1 = (psf[iy+1,ix  ]+psf[iy-1,ix  ]-2*A1)/2
+    E1 = (psf[iy+1,ix+1]-A1)
+    A2 =  psf[iy  ,ix+1]
+    B2 = (psf[iy  ,ix+2]-psf[iy  ,ix  ])/2
+    C2 = (psf[iy+1,ix+1]-psf[iy-1,ix+1])/2
+    D2 = (psf[iy  ,ix+2]+psf[iy  ,ix  ]-2*A2)/2
+    F2 = (psf[iy+1,ix+1]+psf[iy-1,ix+1]-2*A2)/2
+    E2 =-(psf[iy+1,ix  ]-A2)
+    A3 =  psf[iy+1,ix  ]
+    B3 = (psf[iy+1,ix+1]-psf[iy+1,ix-1])/2
+    C3 = (psf[iy+2,ix  ]-psf[iy  ,ix  ])/2
+    D3 = (psf[iy+1,ix+1]+psf[iy+1,ix-1]-2*A3)/2
+    F3 = (psf[iy+2,ix  ]+psf[iy  ,ix  ]-2*A3)/2
+    E3 =-(psf[iy  ,ix+1]-A3)
+    A4 =  psf[iy+1,ix+1]
+    B4 = (psf[iy+1,ix+2]-psf[iy+1,ix  ])/2
+    C4 = (psf[iy+2,ix+1]-psf[iy  ,ix+1])/2
+    D4 = (psf[iy+1,ix+2]+psf[iy+1,ix  ]-2*A4)/2
+    F4 = (psf[iy+2,ix+1]+psf[iy  ,ix+1]-2*A4)/2
+    E4 = (psf[iy  ,ix  ]-A4)
+    V1 = (A1
+        + B1*( fx )
+        + C1*( fy )
+        + D1*( fx )**2
+        + E1*( fx )*( fy )
+        + F1*( fy )**2)
+    V2 = (A2
+        + B2*(fx-1)
+        + C2*( fy )
+        + D2*(fx-1)**2
+        + E2*(fx-1)*( fy )
+        + F2*( fy )**2)
+    V3 = (A3
+        + B3*( fx )
+        + C3*(fy-1)
+        + D3*( fx )**2
+        + E3*( fx )*(fy-1)
+        + F3*(fy-1)**2)
+    V4 = (A4
+        + B4*(fx-1)
+        + C4*(fy-1)
+        + D4*(fx-1)**2
+        + E4*(fx-1)*(fy-1)
+        + F4*(fy-1)**2)
+    rpsf_phot = ((1-fx)*(1-fy)*V1
+               + ( fx )*(1-fy)*V2
+               + (1-fx)*( fy )*V3
+               + ( fx )*( fy )*V4)
+    return rpsf_phot
+
+def place_source(xpix, ypix, flux, image, epsf, boxsize = 44, psf_center = 177):
+    """
+    Place a source into image.
+
+    Parameters
+    ----------
+    xpix : float
+        x location of source
+    ypix : float
+        y location of source
+    flux : float
+        Flux of the source
+    image : numpy.ndarray
+        Empty (or not) 2D array of image where to place sources
+    epsf : numpy.ndarray
+        2D array with ePSF
+
+    Returns
+    -------
+    image : numpy.ndarray
+        2D array of image where sources were placed
+    """
+
+    # Get image size, assuming square
+    image_size = image.shape[0]
+
+    # Apply if source is within image and boxise
+    if (xpix > 0) & (xpix < image_size) & (ypix > 0) & (ypix < image_size):
+        max_y = min(round(ypix+boxsize), image_size)
+        min_y = max(0, round(ypix-boxsize))
+        for j in range(min_y,max_y):
+            dy = j-ypix
+            max_x = min(round(xpix+boxsize), image_size)
+            min_x = max(0, round(xpix-boxsize))
```

Comment:
These can be moved out of the loop since each parameter is fixed for a given call of place_source(), correct?